tests/frontend/unit/ExportModal.spec.js

```diff
@@ -0,0 +1,432 @@
+        },
+        anonymize: true,
+        buttonLabel: 'export.pdf',
+        buttonLabelSingle: 'export.trigger.pdf',
+        exportTypes: true,
+        newPagePerStn: true,
+        obscure: true,
+        tabLabel: 'export.pdf',
+        templates: {
+          condensed: {
+            name: 'export.compact'
+          },
+          landscape: {
+            name: 'export.landscape'
+          },
+          portrait: {
+            name: 'export.portrait'
+          }
+        }
+
+      },
+      xlsx: {
+        _defaults: {
+          anonymous: false,
+          exportType: 'topicsAndTags',
+          sortType: 'false',
+          template: 'compact'
+        },
+        anonymize: false,
+        buttonLabel: 'export.xlsx',
+        buttonLabelSingle: 'export.trigger.xlsx',
+        exportTypes: true,
+        obscure: false,
+        tabLabel: 'export.xlsx',
+      },
+      zip: {
+        _defaults: {
+          anonymous: false,
+          exportType: 'statementsWithAttachments'
+        },
+        buttonLabel: 'export.zip',
+        tabLabel: 'export.zip',
+        templates: {
+          condensed: {
+            name: 'export.compact'
+          },
+          landscape: {
+            name: 'export.landscape'
+          },
+          portrait: {
+            name: 'export.portrait'
+          }
+        }
+      }
+    },
+    procedureId: '1',
+    view: 'assessment_table'
+  }
+
+  let wrapper
+
+  beforeEach(() => {
+    wrapper = shallowMountWithGlobalMocks(
+      ExportModal,
+      {
+        propsData: props
+    })
+  })
+
+  afterEach(() => {
+    wrapper.destroy()
+  })
+
+  it('displays each optGroup as tab', () => {
+    const tabButtons = wrapper.findAll('button')
+    const optGroups = Object.keys(props.options)
+    const buttonTexts = tabButtons.wrappers.map(button => button.text())
+
+    optGroups.forEach(key => {
+      expect(buttonTexts).toContain(props.options[key].tabLabel)
+    })
+  })
+
+  it('switches to the correct tab content when a tab button is clicked', async () => {
+    const tabButtons = wrapper.findAll('button')
+    const index = 1
+    await tabButtons.at(index).trigger('click')
+    const dataCy = tabButtons.at(index).attributes('data-cy')
+    const id = dataCy.split('.').pop()
+    const matchingTabContent = wrapper.find(`#${id}`)
+
+    expect(getComputedStyle(matchingTabContent.element).display).toBe('block')
+  })
+
+    it('triggers a "submit" event when the submit button is clicked', async () => {
+      wrapper.vm.submit = jest.fn()
+      wrapper.vm.$refs.exportModal = {
+        toggle: jest.fn()
+      }
+
+      wrapper.vm.handleSubmit()
+
+      const event = wrapper.emitted('submit')
+
+      expect(wrapper.emitted()).toHaveProperty('submit')
+      expect(event).toHaveLength(1)
+    })
+
+  describe('ExportModal: pdf export', () => {
+    it('displays "anonymized" checkbox in the pdf tab if pdf.anonymize or pdf.obscure option is set to true', async () => {
```

Comment:
Reading this kind of makes me expect to see the setting of one of those props within the test rather than having it set as default but I this is rather my personal preference I guess.

Reply:
Yeah I was wondering the same, but then what should be the default values?

Reply:
I assume setting both to false messes up things for other tests so I have no other idea too.